Add compile-time subgraph schema generation (#1579)
ballerina-platform/module-ballerina-graphql: docs/spec/spec.md

Non-distinct `service` objects and `service` classes can not be used to define or implement GraphQL interfaces.

> >**Note**: Recognition of `service` `object`s and `service` `class`es as GraphQL interfaces and implementations relies on their definition within the same module as the GraphQL service.

Reviewer: On second thoughts, this seems unclear. Shall we reword this? IMO, the spec (or any technical documentation for that matter) should be easy to read and understand. Using excessive adjectives and complex sentences (generated by generative AIs) beats that purpose.

```suggestion
>**Note**: In order to be recognized as GraphQL objects or interfaces, the Ballerina `service` `object`s and `service` `class`es should be defined within the same module as the GraphQL service.
```

> >**Note**: Recognition of `service` `object`s and `service` `class`xes as GraphQL interfaces and implementations relies on their definition within the same module as the GraphQL service.

Author: Updated

> >**Note**: In order to be recognized as GraphQL objects or interfaces, the Ballerina `service` `object`s and `service` `class`es should be defined within the same module as the GraphQL service.

Reviewer: Sorry, mised this.

```suggestion
>**Note**: In order to be recognized as GraphQL objects or interfaces, the Ballerina `service` `object`s and `service` `class`es must be defined within the same module as the GraphQL service.
```

2. Define the `key` field of the annotation to be the fields and subfields that contribute to the entity's primary key/keys.
3. Define the `resolveReference` field of the annotation to be a function pointer to resolve the entity. If this field is set to `nil`, it indicates to the graph router that this subgraph does not define a reference resolver for this entity. For more details, see [ReferenceResolver](#10513-the-subgraphreferenceresolver-function-type).

> >**Note:** To ensure recognition as subgraph entities by the GraphQL module, GraphQL object types marked with `@subgraph:Entity` must be defined within the same Ballerina module as the GraphQL service; failure to comply results in non-recognition.

Reviewer: ```suggestion
>**Note:** In order to be recognized as subgraph entities, GraphQL object types marked with `@subgraph:Entity` must be defined within the same Ballerina module as the GraphQL service.
```
